tests/components/logbook/test_init.py

```diff
@@ -41,264 +40,269 @@
 from tests.common import get_test_home_assistant, init_recorder_component, mock_platform
 from tests.components.recorder.common import trigger_db_commit
 
+EMPTY_CONFIG = logbook.CONFIG_SCHEMA({logbook.DOMAIN: {}})
 
-class TestComponentLogbook(unittest.TestCase):
-    """Test the History component."""
-
-    EMPTY_CONFIG = logbook.CONFIG_SCHEMA({logbook.DOMAIN: {}})
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        init_recorder_component(self.hass)  # Force an in memory DB
-        with patch("homeassistant.components.http.start_http_server_and_save_config"):
-            assert setup_component(self.hass, logbook.DOMAIN, self.EMPTY_CONFIG)
-        self.addCleanup(self.hass.stop)
-
-    def test_service_call_create_logbook_entry(self):
-        """Test if service call create log book entry."""
-        calls = []
-
-        @ha.callback
-        def event_listener(event):
-            """Append on event."""
-            calls.append(event)
-
-        self.hass.bus.listen(logbook.EVENT_LOGBOOK_ENTRY, event_listener)
-        self.hass.services.call(
-            logbook.DOMAIN,
-            "log",
-            {
-                logbook.ATTR_NAME: "Alarm",
-                logbook.ATTR_MESSAGE: "is triggered",
-                logbook.ATTR_DOMAIN: "switch",
-                logbook.ATTR_ENTITY_ID: "switch.test_switch",
-            },
-            True,
-        )
-        self.hass.services.call(
-            logbook.DOMAIN,
-            "log",
-            {
-                logbook.ATTR_NAME: "This entry",
-                logbook.ATTR_MESSAGE: "has no domain or entity_id",
-            },
-            True,
-        )
-        # Logbook entry service call results in firing an event.
-        # Our service call will unblock when the event listeners have been
-        # scheduled. This means that they may not have been processed yet.
-        trigger_db_commit(self.hass)
-        self.hass.block_till_done()
-        self.hass.data[recorder.DATA_INSTANCE].block_till_done()
-
-        events = list(
-            logbook._get_events(
-                self.hass,
-                dt_util.utcnow() - timedelta(hours=1),
-                dt_util.utcnow() + timedelta(hours=1),
-            )
+
+@pytest.fixture
+def hass_():
+    """Set up things to be run when tests are started."""
+    hass = get_test_home_assistant()
+    init_recorder_component(hass)  # Force an in memory DB
+    with patch("homeassistant.components.http.start_http_server_and_save_config"):
+        assert setup_component(hass, logbook.DOMAIN, EMPTY_CONFIG)
+        yield hass
+    hass.stop()
```

Comment:
Make this an async fixture

```suggestion
async def hass_(hass):
    """Set up things to be run when tests are started."""
    await hass.async_add_executor_job(init_recorder_component, hass)  # Force an in memory DB
    with patch("homeassistant.components.http.start_http_server_and_save_config"):
        assert await async_setup_component(hass, logbook.DOMAIN, EMPTY_CONFIG)
    return hass
```

Reply:
Thanks @balloob, I've tried that but then the tests don't running anymore. Gets stuck at the 'collecting' step. Possibly some async process being held but can't figure it out yet.

Reply:
Ah it's probably because all the tests are sync and that breaks things. This is fine then, thanks for taking a look !